src/Exception/InvalidException.php

```diff
@@ -1,26 +1,81 @@
 <?php
 namespace Equip\Auth\Exception;
 
+use Exception;
+
 /**
  * Exception that occurs when a user specifies an authentication token that is
  * invalid.
  */
 class InvalidException extends AuthException
 {
-    const CODE_TOKEN_EXPIRED = 1;
-    const CODE_TOKEN_INVALID = 2;
-    const CODE_CREDENTIALS_INVALID = 3;
+    const CODE = 403;
 
+    /**
+     * @param string $message
+     * @param int $code
+     * @param Exception $previous
+     */
     public function __construct(
         $message = 'The token being used is invalid',
         $code = 0,
-        \Exception $previous = null
+        Exception $previous = null
     ) {
+        if ($code === 0) {
+            $code = static::CODE;
+        }
         parent::__construct($message, $code, $previous);
     }
 
-    public function getStatusCode()
+    /**
+     * @param string $token
+     * @param Exception $previous
+     */
+    public static function tokenExpired($token, Exception $previous = null)
+    {
+        throw new static(
+            sprintf('Token expired: %s', $token),
```

Comment:
And I don't love throwing the value on the end, I generally write it to be a little natural:

```
Provided auth token `%s` is expired
```

etc